datadog/resource_datadog_user.go

```diff
@@ -45,6 +45,7 @@ func resourceDatadogUser() *schema.Resource {
 					Type:        schema.TypeSet,
 					Optional:    true,
 					Elem:        &schema.Schema{Type: schema.TypeString},
+					Deprecated:  "`roles` has been deprecated in favor of the new `datadog_user_role` resource.",
```

Comment:
not deprecated yet! i can imagine customers will still want to use this roles attribute if they want TF to be a source of truth

the `datadog_user_role` resource is meant for incremental adoption, and it won't overwrite non-terraform-managed roles, which might not be ideal

Reply:
ooooops okay!